setup/classes/class.ilObjDefReader.php

```diff
@@ -139,9 +139,9 @@ function handlerBeginTag($a_xml_parser,$a_name,$a_attribs)
 					}
 	
 					$this->current_object = $a_attribs["id"];
-					$ilDB->manipulateF("INSERT INTO il_object_def (id, class_name, component,location,".
+					$ilDB->manipulateF("INSERT INTO il_object_def (id, class_name, `component`,location,".
```

Comment:
Maybe `\ilDBInterface::quoteIdentifier` should be used instead.

Reply:
Or even better $ilDB->insert. Afaik this does quoteIdentifier automatically.